sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -624,7 +624,6 @@ object TypeCoercion {
 
       // Hive lets you do aggregation of timestamps... for some reason
       case Sum(e @ TimestampType()) => Sum(Cast(e, DoubleType))
-      case Average(e @ TimestampType()) => Average(Cast(e, DoubleType))
```

Comment:
When chaning the existing bheivour, we need to update the migration guide and might need to add a legacy config to keep the current behaivour.